From 3ade2dd00d514f6fa345e2acfb3422a8c4efb7a2 Mon Sep 17 00:00:00 2001
From: Allen Wittenauer <aw@apache.org>
Date: Tue, 28 Jun 2016 07:21:04 -0700
Subject: [PATCH 1976/2748] HADOOP-13034. Log message about input options in
 distcp lacks some items (Takashi Ohnishi via aw)

(cherry picked from commit 422c73a8657d8699920f7db13d4be200e16c4272)
(cherry picked from commit a5a56c3564d486f8df258e64ad5c243395def12a)

Change-Id: I77e9cb9be743eb828f900e22538d330be2dcb528
---
 .../org/apache/hadoop/tools/DistCpOptions.java     |    4 ++++
 .../org/apache/hadoop/tools/TestDistCpOptions.java |    1 +
 .../org/apache/hadoop/tools/TestOptionsParser.java |    1 +
 3 files changed, 6 insertions(+)

diff --git a/hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/DistCpOptions.java b/hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/DistCpOptions.java
index b81bc03..a3af917 100644
--- a/hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/DistCpOptions.java
+++ b/hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/DistCpOptions.java
@@ -680,6 +680,10 @@ public String toString() {
         ", deleteMissing=" + deleteMissing +
         ", ignoreFailures=" + ignoreFailures +
         ", overwrite=" + overwrite +
+        ", append=" + append +
+        ", useDiff=" + useDiff +
+        ", fromSnapshot=" + fromSnapshot +
+        ", toSnapshot=" + toSnapshot +
         ", skipCRC=" + skipCRC +
         ", blocking=" + blocking +
         ", numListstatusThreads=" + numListstatusThreads +
diff --git a/hadoop-tools/hadoop-distcp/src/test/java/org/apache/hadoop/tools/TestDistCpOptions.java b/hadoop-tools/hadoop-distcp/src/test/java/org/apache/hadoop/tools/TestDistCpOptions.java
index 5a56cc3..7f15a60 100644
--- a/hadoop-tools/hadoop-distcp/src/test/java/org/apache/hadoop/tools/TestDistCpOptions.java
+++ b/hadoop-tools/hadoop-distcp/src/test/java/org/apache/hadoop/tools/TestDistCpOptions.java
@@ -305,6 +305,7 @@ public void testToString() {
     DistCpOptions option = new DistCpOptions(new Path("abc"), new Path("xyz"));
     final String val = "DistCpOptions{atomicCommit=false, syncFolder=false, "
         + "deleteMissing=false, ignoreFailures=false, overwrite=false, "
+        + "append=false, useDiff=false, fromSnapshot=null, toSnapshot=null, "
         + "skipCRC=false, blocking=true, numListstatusThreads=0, maxMaps=20, "
         + "mapBandwidth=100, sslConfigurationFile='null', "
         + "copyStrategy='uniformsize', preserveStatus=[], "
diff --git a/hadoop-tools/hadoop-distcp/src/test/java/org/apache/hadoop/tools/TestOptionsParser.java b/hadoop-tools/hadoop-distcp/src/test/java/org/apache/hadoop/tools/TestOptionsParser.java
index 33ed37d..a475942 100644
--- a/hadoop-tools/hadoop-distcp/src/test/java/org/apache/hadoop/tools/TestOptionsParser.java
+++ b/hadoop-tools/hadoop-distcp/src/test/java/org/apache/hadoop/tools/TestOptionsParser.java
@@ -400,6 +400,7 @@ public void testToString() {
     DistCpOptions option = new DistCpOptions(new Path("abc"), new Path("xyz"));
     String val = "DistCpOptions{atomicCommit=false, syncFolder=false, "
         + "deleteMissing=false, ignoreFailures=false, overwrite=false, "
+        + "append=false, useDiff=false, fromSnapshot=null, toSnapshot=null, "
         + "skipCRC=false, blocking=true, numListstatusThreads=0, maxMaps=20, "
         + "mapBandwidth=100, sslConfigurationFile='null', "
         + "copyStrategy='uniformsize', preserveStatus=[], "
-- 
1.7.9.5

